src/message.rs

```diff
@@ -8,36 +8,50 @@ use serde::Serialize;
 pub struct Message {
     pub timestamp: DateTime<Utc>,
     pub source: String,
-    pub payload: HashMap<String, PayloadValue>,
+    pub tags: HashMap<String, String>,
+    pub metrics: HashMap<String, PayloadValue>,
 }
 
 impl Message {
-    pub fn new(source: &str) -> Message {
+    pub fn new(source: &str, name: &str, metric: impl Into<PayloadValue>) -> Message {
```

Comment:
How about `metric_name` and `metric_value`

Reply:
Also it is weird that we can insert more than one metric but the constructor only takes one. I guess idea was to only allow one metric name and value per message.

In the case we don't allow multiple metrics per message, we can add `metric_field` and `metric_value` attributes to the struct instead of the hashmap.

In case we allow multiple, how about dropping these arguments here and we'd just use `insert_metric` to add the values and we can make multiple calls as needed.

Reply:
You are right. The constructor is a bit confusing. I want to allow multiple metrics but I also wanted to make sure that it is not possible to create a message without any metrics. This is the only way I could think of that prevents empty messages on compile.

Do you have ideas how I could make sure that message without metrics would never end up in exporter?

Reply:
> Do you have ideas how I could make sure that message without metrics would never end up in exporter?

Not any direct ideas but now that we will make this a getter, we could return an error if metrics is empty. What about that?

Reply:
I can return an error but then it is not possible to prevent creating messages without metrics on compile. I guess there is a trade-off between handling this on compile time or runtime.

I'll change the constructor and panic in getter if metrics is empty.

Reply:
Yep, can't think of any compile-time solution so far.

Reply:
I made the changes